File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        pd = 1
        zero_exist = 0
        for i in nums:
            if i==0:
                zero_exist+=1
            else:
                pd*=i
        if zero_exist>1:
            return [0] * len(nums)
        if zero_exist == 1:
            for i in range(len(nums)):
                if(nums[i]==0):
                    nums[i]=pd
                else:
                    nums[i]=0
                # nums[i]=pd // nums[i]
            return nums
            
        for i in range(len(nums)):
            nums[i]=pd // nums[i]
        return nums
        

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int,int>> pq;
        unordered_map<int,int> mp;
        for(auto & i : nums)
 {
    mp[i]++;
 }
 for(auto [i,j] : mp)
 {
    pq.push({j,i});
 }
 vector<int> ans;
 while(k-->0)
  ans.push_back(pq.top().second),pq.pop();
return ans;

    }
};
